File: src/createEmptyBoard.ts
import { CharacterCode } from "./characterCodes";

export const createEmptyBoard = (
  rows: number,
  columns: number
): Array<number[]> => {
  const result: number[][] = [];

  let i = 0;
  while (i < rows) {
    result[i] = [];
    let i2 = 0;
    while (i2 < columns) {
      result[i][i2] = CharacterCode.Blank;
      i2++;
    }
    i++;
  }

  return result;
};

File: python/pyvbml/create_empty_board.py
"""Create empty board.

Port of Vestaboard/vbml/src/createEmptyBoard.ts
"""

from __future__ import annotations

from .character_codes import CharacterCode


def create_empty_board(rows: int, columns: int) -> list[list[int]]:
    """Create an empty board of *rows* * *columns*."""
    return [[int(CharacterCode.BLANK)] * columns for _ in range(rows)]
